documentation/Database-Schema-Review-And-Consolidations.md

```diff
@@ -0,0 +1,122 @@
+
+`LOQ` -> At least one of applicant's responses marked as non conformity (Decline) and consolidator agreed with reviewers decisions. They have an option to select from LOQ or Non-Conformity
+
+`Non-conofrm (Decline` -> At least one of applicant's responses marked as non conformity (Decline) and consolidator agreed with reviewers decisions. They have an option to select from LOQ or Non-Conformity
+
+`Changes Requested` -> Applies to consolidation, if at least one disagreement with reviewer below consolidator's level, this is the only option available
+
+## Review Assignment
+
+These records are created by back end when review level is reached (either through application submission or review submission, see Review Life Cycle diagram below). `Template Permission` records, alongside `permission name`, `permission name join` and `permission policy` are used to created relevant `review assignments`
+They can be queried to:
+
+- Determine who can be assigned to an application stage and level <- front end task
+- Determine if an application stage and level is fully assigned (using review_question_assignment link) <- through application list view
+- Determine if review can be started/created <- front end can check if review can be created (including self assignment) See `Review and Consolidation URL flow` diagram below
+
+#### Review Assignment Status
+
+`Available` -> can be assigned by assigner
+`Not Available` -> cannot be assigned by assigner, in case of `available for self assignment`, when reviewer self assigns a review, all other `review assignments` for current `stage` and `level` are changed to this status
+`Assigned` -> review is assigned to a user and can be started (in case of self assignment when review is self assigned, review assignment will changes status to assigned)
+`Available for self assignment` -> review can be started and self assigned
+
+`Assignment NOTE` -> For MVP, only level 1 reviewer can be assigned questions (vs self assignment of review), level > 1 will be self assigned (level 1 can also be configured to self assigned, in this case they will need to be configured to review all sections)
+
+##### Other fields (that are not straight away self explanatory)
+
+`assigner_id` -> null until assigned, assigner id when assigned be assigner, otherwise reviewer_id when self assigned
+`is_last_level` -> last level for current stage (determine if review_decision is present in UI)
+`allowable_section_ids` -> an array of section IDs that reviewer has permission to review (would typically have all sections), assigner can only assign questions from sections that are in this list
+
+## Review Question Assignment
+
+Identifies which questions, from the application can be reviewed by the reviewer.
+These records are created upon assignment, and should only be created for level 1 reviewer ? (to confirm)
```

Comment:
Only for level 1? I guess that could work, just wondering which has less convoluted logic. Did we discuss this yet?

Reply:
In the diagrams that were created, this is taken into account (i am not sure if it's demonstrated). 

i.e. level one reviewer responses are created based on review question assignments but lvl +1 review responses are created based on submitted reviews review responses from lvl -1

let me know if you want to chat about this

Reply:
Yea, if we are not creating for other levels we need to be aware that the front-end need to be changed to create the associated review_responses differently. For now it's using the existing `review_question_assignment` to create the `review_responses` here:

https://github.com/openmsupply/application-manager-web-app/blob/d49b5e9fedad4c5b1a7c94aa0a5db63d520dc80b/src/containers/Review/ReviewOverview.tsx#L50